consultation_analyser/support_console/ingest.py

```diff
@@ -54,414 +44,478 @@ def get_all_question_part_subfolders(folder_name: str, bucket_name: str) -> list
-    return import_themes_and_get_framework(question_part=question_part, theme_data=theme_data)
-
-
-def import_all_theme_mappings_from_jsonl(
-    question_part: QuestionPart,
-    framework: Framework,
-    bucket_name: str,
-    question_part_folder_key: str,
-    batch_size: int,
-) -> None:
-    logger.info(
-        f"Importing theme_mappings from {question_part_folder_key}, batch_size {batch_size}"
-    )
-    theme_mappings_file_key = f"{question_part_folder_key}mapping.jsonl"
-    s3_client = boto3.client("s3")
-    response = s3_client.get_object(Bucket=bucket_name, Key=theme_mappings_file_key)
-    lines = []
-    for line in response["Body"].iter_lines():
-        lines.append(line)
-        if len(lines) == batch_size:
-            import_theme_mappings_job.delay(
-                question_part=question_part, thememapping_data=lines, framework=framework
-            )
-            lines = []
-    # Any remaining lines < batch size
-    if lines:
-        import_theme_mappings_job.delay(
-            question_part=question_part, thememapping_data=lines, framework=framework
-        )
-
-
-def import_all_sentiment_mappings_from_jsonl(
-    question_part: QuestionPart, bucket_name: str, question_part_folder_key: str, batch_size: int
-) -> None:
-    logger.info(
-        f"Importing sentiment mappings from {question_part_folder_key}, batch_size {batch_size}"
-    )
-    sentiment_mappings_file_key = f"{question_part_folder_key}sentiment.jsonl"
-    s3_client = boto3.client("s3")
-    response = s3_client.get_object(Bucket=bucket_name, Key=sentiment_mappings_file_key)
-    lines = []
-    for line in response["Body"].iter_lines():
-        lines.append(line)
-        if len(lines) == batch_size:
-            import_sentiment_mappings_job.delay(
-                question_part=question_part, sentimentmapping_data=lines
+            respondent_count += 1
+        
+        Respondent.objects.bulk_create(respondents_to_save)
+        logger.info(f"Imported {respondent_count} respondents")
+        
+        # 3. Process each question
+        question_folders = get_question_folders(inputs_path, bucket_name)
+        
+        for question_folder in question_folders:
+            question_num_str = question_folder.split("/")[-2].replace("question_part_", "")
+            question_number = int(question_num_str)
+            
+            logger.info(f"Processing question {question_number}")
+            
+            # 3a. Import question
+            question_file_key = f"{question_folder}question.json"
+            response = s3_client.get_object(Bucket=bucket_name, Key=question_file_key)
+            question_data = json.loads(response["Body"].read())
+            
+            question_text = question_data.get("question_text", "")
+            if not question_text:
+                raise ValueError(f"Question text is required for question {question_number}")
+            
+            question = Question.objects.create(
```

Comment:
Assume we're doing ingest of multiple choice separately.

Reply:
All sounds sensible - also probably shouldn't increase the scope of this PR any more!